src/python/pants/task/simple_codegen_task.py

```diff
@@ -230,7 +234,8 @@ def execute(self):
       with self.context.new_workunit(name='execute', labels=[WorkUnitLabel.MULTITOOL]):
         vts_to_sources = OrderedDict()
         for vt in invalidation_check.all_vts:
-          synthetic_target_dir = self.synthetic_target_dir(vt.target, vt.results_dir)
+
+          synthetic_target_dir = self.synthetic_target_dir(vt.target, vt.current_results_dir)
```

Comment:
What's the difference between results_dir and current_results_dir? Why is this changing?

Reply:
There was a bug in this area, since `go` was actually overriding `synthetic_target_dir`. https://github.com/pantsbuild/pants/pull/7241/commits/d934b2b8dce181aa328103cbeaf479969ad7e9a6 makes the usage and explanation-of-usage of these methods more consistent.